netket/driver/vmc_rgn.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import jax.numpy as jnp
+
+from textwrap import dedent
+
+from netket.utils.types import PyTree, Array
+from netket.operator import AbstractOperator
+from netket.stats import Stats
+from netket.vqs import MCState
+from netket.optimizer import (
+    identity_preconditioner,
+    PreconditionerT,
+)
+from netket.utils import warn_deprecation
+from netket.optimizer import Sgd
+
+from .vmc_common import info
+from .abstract_variational_driver import AbstractVariationalDriver
+
+
+class VMC_RGN(AbstractVariationalDriver):
+    """
+    Energy minimization using Variational Monte Carlo (VMC).
+    """
+
+    # TODO docstring
+    def __init__(
+        self,
+        hamiltonian: AbstractOperator,
+        schedule: Array,
+        optimizer=Sgd(learning_rate=1),
+        variational_state=None,
+        preconditioner: PreconditionerT = None,
+        sr: PreconditionerT = None,
+        sr_restart: bool = None,
+        **kwargs,
+    ):
+        """
+        Initializes the driver class.
+
+        Args:
+            hamiltonian: The Hamiltonian of the system.
+            optimizer: Determines how optimization steps are performed given the
+                bare energy gradient.
+            preconditioner: Determines which preconditioner to use for the loss gradient.
+                This must be a tuple of `(object, solver)` as documented in the section
+                `preconditioners` in the documentation. The standard preconditioner
+                included with NetKet is Stochastic Reconfiguration. By default, no
+                preconditioner is used and the bare gradient is passed to the optimizer.
+        """
+
+        if variational_state is None:
```

Comment:
Assuming you change the inheritance above, you can replace this whole method with
```suggestion
        super().__init__(*args, **kwargs)
        self.schedule = schedule
```